plugins/removeHiddenElems.js

```diff
@@ -356,37 +394,35 @@ exports.fn = (root, params) => {
           removeElement(node, parentNode);
           return;
         }
-
-        if (node.name === 'use') {
-          const referencesRemovedDef = Object.entries(node.attributes).some(
-            ([attrKey, attrValue]) =>
-              (attrKey === 'href' || attrKey.endsWith(':href')) &&
-              removedDefIds.has(
-                attrValue.slice(attrValue.indexOf('#') + 1).trim()
-              )
-          );
-
-          if (referencesRemovedDef) {
-            detachNodeFromParent(node, parentNode);
+      },
+    },
+    root: {
+      exit: () => {
+        // Remove uses of deleted definitions
+        for (const id of removedDefIds) {
+          const refs = referencesById.get(id);
+          if (refs) {
+            for (const { node, parentNode } of refs) {
+              detachNodeFromParent(node, parentNode);
+            }
           }
-
-          return;
         }
 
-        if (node.name === 'svg' && parentNode.type === 'root') {
+        // Remove definitions that are unused
+        if (!deoptimized) {
```

Comment:
Could you provide an example/test where de-optimizing for styles makes sense?

(This section needs a refactor in general, we should be collecting the relevant data while visiting instead of CSS selecting at the end. I can handle this later, since it's out-of-scope for this PR, though.)

Reply:
Something like this
```xml
<svg>
  <defs>
    <linearGradient id="a">
      ...
    </linearGradient>
  </defs>
  <rect width="128" height="128></rect>
  <style> rect { fill: url(#a); } </style>
</svg>
```

Reply:
Make sense! I forgot definitions could be referenced from the style element.

I'd like if we could remove the exception on styles later, but this is fine for now. Later I can work on a PR to check the stylesheet for references to the ID as well.